src/com/owncloud/android/operations/SynchronizeFolderOperation.java

```diff
@@ -165,13 +165,25 @@ protected RemoteOperationResult run(OwnCloudClient client) {
                     sendBroadcastForNotifyingUIUpdate(result.isSuccess());
                 }
             }
-        } catch (OperationCancelledException e) {
-            result = new RemoteOperationResult(e);
             
-            // cancel 'child' synchronizations
-            for (SyncOperation  synchOp: mFoldersToWalkDown) {
-                ((SynchronizeFolderOperation) synchOp).cancel();
+            if (mCancellationRequested.get()) {
+                throw new OperationCancelledException();
             }
+            
+        } catch (OperationCancelledException e) {
+            result = new RemoteOperationResult(e);
+
+            // Needed in case that cancellation occurs before starting any download.
+            // If not, yellow arrow continues being shown.
+            sendBroadcastForNotifyingUIUpdate(result.isSuccess());
+
+            /// cancellation of download needs to be done separately in any case; a SynchronizeFolderOperation
```

Comment:
This comment should be removed, it's copied from ```OperationsService```; it makes sense there, not here (my fault)